Quadriga/src/main/java/edu/asu/spring/quadriga/rest/NetworkRestController.java

```diff
@@ -48,661 +50,636 @@
+
+            if (networkId.endsWith(INetworkManager.DSPACEERROR)) {
+                String errorMsg = "Text files don't belong to this workspace.";
+                return new ResponseEntity<String>(errorMsg, HttpStatus.NOT_FOUND);
+            }
+
+            // TODO: Send email to all editors of a workspace
+            // TODO: Get the workspace editors from the workspaceid
+
+            HttpHeaders httpHeaders = new HttpHeaders();
+            httpHeaders.setContentType(MediaType.valueOf(accept));
+            httpHeaders.set("networkid", networkId);
+            return new ResponseEntity<String>(res, httpHeaders, HttpStatus.OK);
+        }
+
+    }
+
+    /**
+     * Rest interface for getting status of network http://<<URL>:
+     * <PORT>>/quadriga/rest/networkstatus/{NetworkId}
+     * http://localhost:8080/quadriga/rest/networkstatus/NET_1huxp4w7p71o0
+     * 
+     * @param networkName
+     * @param response
+     * @param accept
+     * @param principal
+     * @return status
+     * @throws Exception
+     */
+    @RequestMapping(value = "rest/networkstatus/{NetworkId}", method = RequestMethod.GET, produces = "application/xml")
+    public ResponseEntity<String> getNetworkStatus(@PathVariable("NetworkId") String networkId,
+            HttpServletResponse response, String accept, Principal principal, HttpServletRequest req) throws Exception {
+        INetwork network = networkManager.getNetwork(networkId);
+        if (network == null) {
+            String errorMsg = errorMessageRest.getErrorMsg("Network ID : " + networkId + " doesn't exist");
+            return new ResponseEntity<String>(errorMsg, HttpStatus.NOT_FOUND);
+        }
+        IUser user = userManager.getUser(principal.getName());
+        VelocityEngine engine = restVelocityFactory.getVelocityEngine(req);
+        Template template = null;
+        String status = networkManager.getNetworkStatusCode(network.getStatus()) + "";
+        try {
+            engine.init();
+
+            template = engine.getTemplate("velocitytemplates/networkstatus.vm");
+            VelocityContext context = new VelocityContext(restVelocityFactory.getVelocityContext());
+            if (network.getStatus() == null)
+                context.put("status", INetworkStatus.UNKNOWN_CODE + "");
+            else
+                context.put("status", status);
+            context.put("networkid", networkId);
+            List<INetworkAnnotation> networkAnnoList = editingNetworkAnnoManager
+                    .getAllAnnotationOfNetwork(user.getUserName(), networkId);
+            context.put("networkAnnoList", networkAnnoList);
+            StringWriter writer = new StringWriter();
+            template.merge(context, writer);
+
+            HttpHeaders httpHeaders = new HttpHeaders();
+            httpHeaders.setContentType(MediaType.APPLICATION_XML);
+            return new ResponseEntity<String>(writer.toString(), httpHeaders, HttpStatus.OK);

```

Comment:
move everything out of this try block and only put try catch around the code that actually throws the exception (e.g. building the context can probably be done before the try).